internal/tofu/node_resource_abstract_instance.go

```diff
@@ -346,6 +346,55 @@ func (n *NodeAbstractResourceInstance) writeResourceInstanceStateImpl(ctx EvalCo
 	return nil
 }
 
+// planForget returns a removed from state diff.
+func (n *NodeAbstractResourceInstance) planForget(ctx EvalContext, currentState *states.ResourceInstanceObject, deposedKey states.DeposedKey) (*plans.ResourceInstanceChange, tfdiags.Diagnostics) {
+	var diags tfdiags.Diagnostics
+	var plan *plans.ResourceInstanceChange
+
+	absAddr := n.Addr
+
+	// If there is no state or our attributes object is null then the resource
+	// was already forgotten.
+	if currentState == nil || currentState.Value.IsNull() {
+		// We still need to generate a NoOp change, because that allows
+		// outside consumers of the plan to distinguish between us affirming
+		// that we checked something and concluded no changes were needed
+		// vs. that something being entirely excluded e.g. due to -target.
+		noop := &plans.ResourceInstanceChange{
+			Addr:        absAddr,
+			PrevRunAddr: n.prevRunAddr(ctx),
```

Comment:
Just for clarification, can we rename AND remove at the same time? This feels a little complicated. If we didn't have to handle renames and deposed keys it would be simpler.

Reply:
> Just for clarification, can we rename AND remove at the same time? 

That's a good question. I had a todo to check all possible moved block options with the removed block, and I found one scenario in which you can move and delete at the same time.
From this situation:
```
resource "null_resource" "example_resource" {
   provisioner "local-exec" {
     command = "echo 'Hello from null resource'"
   }
 }
```
To this:
```
removed {
   from = null_resource.example_resource2
}

moved {
   from = null_resource.example_resource
   to = null_resource.example_resource2
}

/* resource "null_resource" "example_resource" {
   provisioner "local-exec" {
     command = "echo 'Hello from null resource'"
   }
 } */

  /* resource "null_resource" "example_resource2" {
	   provisioner "local-exec" {
	     command = "echo 'Hello from null resource'"
	   }
   } */
```

Reply:
Also, I added a test for the `moved` and the `removed` blocks at the same time - [here](https://github.com/opentofu/opentofu/pull/1158/files#diff-29c9b1b9ff61214376c78ddd5be7b929d5864b8f131aa9cd386c56c9585c0a2fR5807).